### PR TITLE
VCR integration

### PR DESCRIPTION
**Pull request by beingmattlevy**
Initial checkin
payment spec first cassettes

### CODE REVIEW

File: app/models/payment.rb [hunk 1]
```diff
@@ -221,6 +217,24 @@ def convert_old_status!
...
+  # Manually mark that a payment was received.
+  def mark_received
+    status_received!
+  end
+
+  # In transaction,
+  # mark payment received and ticket request completed
+  # Send off PaymentMailer for payment received
+  def request_completed
+    mark_received
```
Comment (kigster) [inline app/models/payment.rb:229]: Do we want to wrap these two in a transaction?

**Review by kigster — CHANGES_REQUESTED** [review]
Awesome work man! Tests now pass in 6 seconds!!! Whoohoo!

Can you add to README instructions on how to regenerate cassettes if any of the APIs change? Or how to turn off using cassettes and turn on cassette generation?

I left a couple of tiny questions but this is nearly ready to go.

File: app/models/payment.rb (hunk 1 above)
Comment (beingmattlevy) [reply app/models/payment.rb:229]: I'll move the transaction from the controller.

File: app/controllers/payments_controller.rb [hunk 2]
```diff
@@ -103,7 +103,7 @@ def sent
   private
 
   def mark_payment_completed
-    Payment.transaction do
```
Comment (kigster) [inline app/controllers/payments_controller.rb:106]: No need for begin if it follows method entry. You can add rescue to any method without a begin.

File: app/models/payment.rb [hunk 3]
```diff
@@ -222,17 +222,17 @@ def mark_received
...
 
-  # In transaction,
   # mark payment received and ticket request completed
   # Send off PaymentMailer for payment received
   def request_completed
-    mark_received
-    ticket_request.mark_complete
+    Payment.transaction do
+      # set payment and ticket request status
+      mark_received
```
Comment (kigster) [inline app/models/payment.rb:230]: 👍

**Review by kigster — APPROVED** [review]
Approving, minus the extraneous begin/rescue in a method.

any method entry works just like `begin` and so you can rescue any exception within the method without the explicit and unnecessary `begin`

**Review by kigster — APPROVED** [review]

**Pull request merged**